src/lib/util/RunOutputProcessor.ts

```diff
@@ -90,7 +93,7 @@ export class RunOutputProcessor {
 			msgParts = [...msgParts, ...summaryMsgs];
 		}
 		// If we're supposed to throw an exception in response to violations, we need an extra piece of summary.
-		if (minSev > 0 && this.opts.violationsCauseException) {
+		if ((minSev > 0 && this.opts.violationsCauseException) || (this.opts.severityForError && minSev <= this.opts.severityForError)) {
```

Comment:
Another nitpick: How about extracting the `severityForError` logic to a separate method? Will help improve code readability. Also, would be useful to have a comment explaining why we do a `<=` here. If we decide to modify the severity from ascending to descending, this information would be helpful.